data/Record.js

```diff
@@ -52,6 +52,9 @@ export class Record {
         return this.store.getChildrenById(this.id, false);
     }
 
+    get isSummary() {
+        return this.id === 'summary';
```

Comment:
I don't think we want to require a special id.   The id's of our root nodes can be significant.  Let's discuss whether it would be ok to assume that a single row with zero siblings is the summary?

Reply:
tl;dr - I ran into complications trying to use the shape of the data to identify the summary record, so went down the path of making it explicitly identified.

I was initially handling it the way you suggest, with treating a root node with no siblings as the "summary" record, but that really didn't work out well in all cases.

In the case that the data itself is not hierarchical I didn't want to force apps to add a root node as the means of supplying a summary record, though I could be persuaded on this point.

Also in order to make sure the summary record itself is handled correctly when filtering, I am keeping it completely separated from the main record list/map. This seemed like the cleanest solution for this, and didn't require that additional logic be added in various methods in RecordSet to handle the summary record. To have things work that way it is necessary to know exactly which record is the summary record when the data is processed by RecordSet, and since it is entirely possible that the data being loaded into the store has a single root node which is _not_ a summary record, it is necessary to be explicit when identifying the summary record instead of following some kind of convention based on the shape of the data.

I see a few options here for explicitly identifying the summary record:

1. Have a "special" id we use to identify, but make this something that can be configured on the Store (`summaryIdSpec`?). This would allow the app to keep their root ids. We could default it to `xhSummaryRecordId` or something and have a constant to clean up the code and make it easy for apps to access it. 
2. Instead of using the id, we could look for an `isSummary` prop on the raw data to identify the summary record. This would mean reserving another prop on Record that apps couldn't use for fields, and wanted to avoid doing that, so I went with the id approach.
3. Separate the summary data even more from the other data by having a `setSummaryData` or `loadSummaryData` on Store, and never even keep it with the other records. Would require either having a separate RecordSet for the summary record or some refactoring would be needed to be able to create the actual record without going through a RecordSet.

Please let me know if I am over-complicating things here :) Would be happy to have a simpler means of managing the summary record.